### DISCUSSION TITLE
How do I move polygons made in a loop? (Hexagon Layout)

### DISCUSSION THREAD

**Discussion by SigmaXD** [spakin/SimpInkScr#123] [Q&A]
Hi!

I'm making a script that will hopefully layout a hexagon grid. The code, so far, is provided below. 
The three "for" loops are used to create the hexagons. The q, r, s, variables serve as the offset from the origin hexagon at (0, 0, 0). Increasing the range in the loops would create more hexagons but I don't know how to use the transform functions. 

Currently, I'm passing a lot of variables into the draw_hex() function. For example, a draw_hex(1, 2, -1, x, y, 30, points) call will eventually offset that specific hexagon along the the q, r, s axis by the double size of the radius multiplied by the number: move up the q-axis by one, move up the r-axis by two. move down the s-axis by one.

```
import math

def main():
  #initial HEX coordinates
  q, r, s = 0, 0, 0
  #initial 2D coordinates (python)
  x, y = 0, 0
  #initial 2D coordinates (inkscape)
  x, y = canvas.width/2, canvas.height/2
  #size / scale
  radius = 30
  #scale = 1
  
  for q in range(0, 1):
    for r in range(0, 1):
      for s in range(0, 1):
        points = hexagon_vertices(x, y, radius)
        draw_hex(q, r, s, x, y, radius, points)


def hexagon_vertices(x, y, radius):
    # Calculate the six corners of the hexagon from end point     
    points = []
    
    #Flat Top Hex
    for i in range(6):
        angle = i * math.pi / 3 # 60 degree increments
        new_x = round(x + radius * math.cos(angle), 5)
        new_y = round(y + radius * math.sin(angle), 5)
        points.append((new_x, new_y))

    return points

def draw_hex(q, r, s, x, y, radius, points):
   polygon(points)


if __name__ == '__main__':
    main()

```

I tried this variation to move the hexagon but received errors when trying to run through Inkscape. Any help on correcting the following function would be greatly appreciated. 

```
def draw_hex(q, r, s, x, y, radius, points):

  hex = polygon(points)
  x = x * q
  y = y * r
  hex.translate((x, y))

```

**Comment by spakin** [comment]
What errors are you receiving?  Your modified `draw_hex` executes without error when I run it from the Inkscape GUI.  I tested your code with Inkscape 1.2.2 (b0a8486541, 2022-12-01) and Inkscape 1.3.2 (091e20e, 2023-11-25), both on Ubuntu Linux.

P.S.  You may be interested in Simple Inkscape Scripting's [`regular_polygon`](https://github.com/spakin/SimpInkScr/wiki/Shape-construction#polygons) function, which will let you replace both your `hexagon_vertices` and `draw_hex` functions with a single expression like `regular_polygon(6, (x + x*q, y + y*r), radius, math.pi/3)`.
